lxd/endpoints/network.go

```diff
@@ -164,8 +185,9 @@ func networkCreateListener(address string, cert *shared.CertInfo) (net.Listener,
 // continue using the old configuration.
 type networkListener struct {
 	net.Listener
-	mu     sync.RWMutex
-	config *tls.Config
+	mu           sync.RWMutex
+	config       *tls.Config
+	trustedProxy string
```

Comment:
I think it'd be more efficient if `trustedProxy` was a `[]net.IP`.

So you'd do the parsing of the string once in `NetworkUpdateTrustedProxy` and then record the list of proxy servers as Go IP addresses making `isProxy` much cheaper.